ql/src/java/org/apache/hadoop/hive/ql/optimizer/stats/annotation/StatsRulesProcFactory.java

```diff
@@ -2553,7 +2553,10 @@ private void updateColStats(HiveConf conf, Statistics stats, long leftUnmatchedR
             = HiveConf.getBoolVar(conf, ConfVars.HIVE_STATS_JOIN_NDV_READJUSTMENT);
         long newDV = oldDV;
         if (useCalciteForNdvReadjustment) {
-          newDV = RelMdUtil.numDistinctVals(oldDV * 1.0, newNumRows * 1.0).longValue();
+          Double approxNdv = RelMdUtil.numDistinctVals(oldDV * 1.0, newNumRows * 1.0);
+          if (approxNdv != null) {
```

Comment:
We should probably add a precondition so we fail it if is null. Otherwise, we will basically not scale the value silently, which is neither the new behavior or the old behavior.